src/components/multiSelect/MultiSelectBottomSheet.tsx

```diff
@@ -52,25 +52,41 @@ function MultiSelectBottomSheet<T extends string>({
   const isEveryOptionSelected = options.length === selectedOptions.length
 
   const handleClose = () => {
-    onClose?.()
+    forwardedRef.current?.close()
+  }
+
+  const handleSelection = (newOptions: T[]) => {
+    setSelectedOptions(newOptions)
+    onSelect?.(newOptions)
+  }
+
+  const handleSelectAll = () => {
+    handleSelection(options.map((option) => option.id))
+
+    if (mode === 'select-all-or-one') {
+      handleClose()
+    }
   }
 
   const toggleOption = (option: Option<T>) => {
-    setSelectedOptions((prevSelectedOptions) => {
-      if (mode === 'select-all-or-one' || options.length === prevSelectedOptions.length) {
-        return [option.id]
-      } else if (prevSelectedOptions.includes(option.id)) {
-        return prevSelectedOptions.filter((selectedOption) => selectedOption !== option.id)
-      } else {
-        return [...prevSelectedOptions, option.id]
-      }
-    })
+    if (mode === 'select-all-or-one') {
+      // automatically close the bottom sheet after state update if only one option can be selected
+      handleSelection([option.id])
+      handleClose()
+    } else {
+      const isOptionPreviouslySelected = selectedOptions.includes(option.id)
```

Comment:
oh yeah, that is strange :/ i wasn't involved in the design or initial implementation of this component, so can't comment about expected behaviour but i kind of suspect that it's just a detail that hasn't been ironed out because we haven't applied this multiple select variation in a real design yet...rather than spend time on it now, i kind of want to punt this to later (maybe by the time we need it IRL, this component has evolved)